### ISSUE TITLE
Wayland window dialogues expiring instantly

### ISSUE THREAD

**Issue by ooliver1** [godotengine/godot#82205]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - Ubuntu 23.04 23.04 - Vulkan (Forward+) - integrated Intel(R) Graphics (RKL GT1) () - 11th Gen Intel(R) Core(TM) i5-11400 @ 2.60GHz (12 Threads)

### Issue description

https://cdn.discordapp.com/attachments/1155205406440833084/1155205585755713709/Screencast_from_2023-09-23_19-12-50.webm

Mini dialogues like add node sometimes disappear and either have to be re-opened or refuse to float to the top. The following logs support this:

```
gnome-shell[1454] Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0xe00004
gnome-shell[1454] Invalid WM_TRANSIENT_FOR window 0x0 specified for 0xe00325.
```

### Steps to reproduce

1. Create a scene
2. Open the new node dialogue with ctrl + A
3. Exit this and retry step 2 until this happens (does not happen all the time, mostly fine after a restart for a little while)

### Minimal reproduction project

This is project-agnostic, happens on any project (including godot examples). I understand how little this issue provides in repro steps but it's worth a try.

**Comment by danilw** [comment]
If you mean - window just "hide" and you can not alt-tab on it but you see it on Gnome overview - same happening to me on Wayland with Godot 4 - very randomly and very annoying.

> Open the new node dialogue with ctrl + A
> Exit this and retry step 2 until this happens

**from my experience with this Editor behavior** - it happening only when you select "text field" on creating new window or even opening GDScript/Shader editor

**there is way to restore window**, without closing editor - you need to "switch focus" from text-field element to any else, and you have 1 frame to do it - on alt-tab to this "hidden-bugged window" it show itself for 1 frame - and if you click on some other UI element for this 1 frame - bug gone and you can again use this window.

**My current solution** - `Editor -> Editor Settings -> Interface-Editor-> check Single Window mode` - setting this solve this behavior for me.

**Comment by ooliver1** [comment]
Yes that's it, it didn't happen all too often but happens a lot more, but that was also before updating to 4.1.

Side note, wayland is just the protocol so you can't have a wayland bug persay, in my case it would be a mutter bug, but whatever window manager you use. If you use a different window manager then it's either xwayland or godot itself.

**Comment by Chaosus** [comment]
Waiting until the https://github.com/godotengine/godot/pull/57025 got merged has a meaning for this issue, I guess.

**Comment by Calinou** [comment]
Thanks for the report! Consolidating in https://github.com/godotengine/godot/issues/77333, which is fixed in 4.2.dev5.
